gossip/gasprice/reactive.go

```diff
@@ -261,8 +261,9 @@ func (gpo *Oracle) calcTxpoolStat() txpoolStat {
 	p := uint64(0)
 	for _, tx := range sorted {
 		for p < uint64(len(s.percentiles)) && gasCounter >= p*maxGasToIndex/uint64(len(s.percentiles)) {
-			s.percentiles[p] = tx.EffectiveGasTipValue(minGasPrice)
-			if s.percentiles[p].Sign() < 0 {
+			gasTip, err := tx.EffectiveGasTip(minGasPrice)
+			s.percentiles[p] = gasTip
```

Comment:
This does not look right. If there is an error, `gasTip` should not be used, right?